app/commands/incident.py

```diff
@@ -31,6 +32,16 @@ def handle_incident_action_buttons(client, ack, body, logger):
         body["original_message"]["attachments"] = attachments
         body["original_message"]["channel"] = body["channel"]["id"]
 
+        # rich_text blocks are only available for 1st party Slack clients (meaning Desktop, iOS, Android Slack apps)
+        # https://github.com/slackapi/bolt-js/issues/1324
+        if "blocks" in body["original_message"]:
+            for block in body["original_message"]["blocks"]:
+                if "type" in block and block["type"] == "rich_text":
+                    delete_block = True
+
+        if delete_block:
```

Comment:
Question: do we have to delete all "blocks" if there's one `rich_text` block found?  Or could we only remove the `rich_text` blocks?

And follow-up, would that even make sense - perhaps selectively deleting blocks would result in a garbled message.

Reply:
From the linked issue they mention that as long as have valid new text when calling the update api, the original message will persist when you delete the block contents. My original thought was to remove only the rich_text but as you said it'd result in partial results. I'd rather have nothing vs incomplete 😂 . The below is how it looks like when you remove the content inside the block.

![image](https://user-images.githubusercontent.com/85885638/167755649-83e20e77-19ba-416a-8888-8a9d759a4542.png)

Reply:
That makes sense - and yeah, it's a good point to have nothing instead of incomplete.